### DISCUSSION TITLE
Active tracks

### DISCUSSION THREAD

**Discussion by tdelabro** [keep-starknet-strange/madara#1288] [General]
# What is going on in Madara right now?

### Goal
l1->l2: Madara's client listen to event from the Starknet core contracts (deployed on some test foundry network) and push the corresponding transaction in the mempool, then they are executed by the runtime.
l2->l1: Madara's client react to new block finalization by extracting the messages contained in this block's transactions and submitting the corresponding Ethereum transaction on l1 (the foundry network)
### People involved
Timothée (exploration team): gh @tdelabro, tg: @timothee_delabrouille
Michael: gh: @m-kus, tg: @m_kus
Daniel: gh: @bidzyyys,  tg: @bidzyyys
### Related work
#1132 #1133  #1166

### Goal 
To be able to prove the a block we first need to re-execute it in the StarkneOS (SNOS), a cairo0 program with a lot of complex hints. Starknet mainet uses python hints, we want to have those written in Rust.
### People involved
Ben (exploration team): gh: @drspacemn, tg: @drspacemn
Lucas  (exploration team): gh: @LucasLvy, tg: @lucasLvy
Greg (Kakarot): gh: @greged93, tg: @greged93
### Related work
https://github.com/keep-starknet-strange/snos

### Goal
For a lot of reason, the Starknet protocol include in each block header different commitments. One of those is the global_state_root which is the root of a really big trie representing all of the chain state. We have to find the best way to compute it efficiently.
### People involved
Timothée (exploration team): gh @tdelabro, tg: @timothee_delabrouille
Antiryo (Kasar labs): gh: @antiyro , tg: @antiyro
Mathias (Pragma): gh: @EvolveArt, tg: @evolve0x

### Goal
As Madara is heading towards production, we need better docs, websites, library compatibility and tutorials on how new projects can build on top of Madara and whether they should do so in the first place. The goal of this track is start building these tools to make the DX as easy as possible.
### People involved
Apoorv (Karnot): gh @apoorvsadana, tg: @apoorvsadana
Matteo (Pragma): tg: @BGLabs
Lana (exploration team): gh @lana-shanghai, tg: @Edouard_M
Edouard (Edouard_M): tg: @Edouard_M

**Comment by antiyro** [comment]
## P2P Starknet Architecture

### Goal

To ensure 100% compatibility with Starknet, it is essential for Madara to support the peer-to-peer (P2P) specifications determined by the StarkNet ecosystem, as outlined at https://github.com/starknet-io/starknet-p2p-specs. The goal is to build them upon the current specifications of Substrate.

## People involved

Antiyro (Kasar): GitHub @antiyro, Telegram @antiyro
Contributions accepted 👍

**Reply by tdelabro** [reply]
Can you come up with a tracking issue listing the work that need to be done. This way we can dispatch it between contributors?

**Reply by antiyro** [reply]
Yes will do so when we finish: https://github.com/keep-starknet-strange/madara/discussions/1288#discussioncomment-7801035

**Comment by antiyro** [comment]
## Maintain Starknet RPC compatibility

### Goal

To ensure 100% compatibility with Starknet, it is essential for Madara to support the latests versions of Starknet JSON-RPC specs referenced [here](https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json). Currently Madara is under [v0.4.0](https://github.com/starkware-libs/starknet-specs/releases/tag/v0.4.0) and an effort has already started regarding [v0.5.1](https://github.com/starkware-libs/starknet-specs/releases/tag/v0.5.1) the latest stable version [here](https://github.com/keep-starknet-strange/madara/issues/1291)

### People involved

Antiyro (Kasar): GitHub @antiyro, Telegram @antiyro
Contributions accepted 👍

**Reply by antiyro** [reply]
https://github.com/keep-starknet-strange/madara/pull/1313 implements v0.5.1

**Reply by antiyro** [reply]
Starting v0.6.0 implementation.

**Comment by antiyro** [comment]
## Maintain Starknet overall compatibility

### Goal

To ensure 100% compatibility with Starknet, it is essential for Madara to support the latests versions of Starknet referenced [here](https://docs.starknet.io/documentation/starknet_versions/version_notes/). Currently Madara is under [v0.12.0](https://docs.starknet.io/documentation/starknet_versions/version_notes/#version0.12.0) and an effort should start in the upcoming week regarding [v0.13.0](https://docs.starknet.io/documentation/starknet_versions/version_notes/#version0.13.0) the latest stable version

### People involved

Antiyro (Kasar): GitHub @antiyro, Telegram @antiyro
Contributions accepted 👍
